app/main.py

```diff
@@ -2,0 +1,43 @@
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        if self.health >= 0:
+            self.alive = True
+            Animal.alive.append(self)
```

Comment:
Appending `self` to `Animal.alive` here assumes that `alive` is a class variable. If you change `alive` to an instance variable, you need to adjust this logic accordingly.